### DISCUSSION TITLE
How to handle panic in axum? I didn't find anything about this in the examples.

### DISCUSSION THREAD

**Discussion by rfvygv** [tokio-rs/axum#1865] [Q&A] [answered]
How to catch panic and return a json response? Please add an example.

**Comment by syphar** [comment, marked as answer]
There is [`CatchPanicLayer`](https://docs.rs/tower-http/latest/tower_http/catch_panic/index.html) from `tower-http`, which also includes an example of how to return a custom response.

**Reply by rfvygv** [reply]
Thank you so much.
